Tests/SwiftFormatRulesTests/DoNotUseSemicolonsTests.swift

```diff
@@ -18,4 +18,79 @@ public class DoNotUseSemicolonsTests: DiagnosingTestCase {
                 print("3")
                 """)
   }
+
+  public func testSemicolonsInNestedStatements() {
+    XCTAssertFormatting(
+      DoNotUseSemicolons.self,
+      input: """
+             guard let someVar = Optional(items.filter ({ a in foo(a); return true; })) else {
+               items.forEach { a in foo(a); }; return;
+             }
+             """,
+      // The formatting in the expected output is unappealing, but that is fixed by the pretty
+      // printer and isn't a concern for the format rule.
+      expected: """
+                guard let someVar = Optional(items.filter ({ a in foo(a)
+                return true})) else {
+                  items.forEach { a in foo(a)}
+                return
+                }
+                """)
+  }
+
+  public func testSemicolonsInMemberLists() {
+    XCTAssertFormatting(
+      DoNotUseSemicolons.self,
+      input: """
+             struct Foo {
+               func foo() {
+                 code()
+               };
+
+               let someVar = 5;let someOtherVar = 6;
+             }
+             """,
+      expected: """
+                struct Foo {
+                  func foo() {
+                    code()
+                  }
+
+                  let someVar = 5
+                let someOtherVar = 6
+                }
+                """)
+  }
+
+  public func testNewlinesAfterSemicolons() {
+    XCTAssertFormatting(
+      DoNotUseSemicolons.self,
+      input: """
+             print("hello");
+             /// This is a doc comment for printing "goodbye".
+             print("goodbye");
+
+             /// This is a doc comment for printing "3".
+             print("3");
+
+             print("4"); /** Inline comment. */ print("5");
+
+             print("6");  // This is an important statement.
+             print("7");
+             """,
+      expected: """
+                print("hello")
+                /// This is a doc comment for printing "goodbye".
+                print("goodbye")
+
+                /// This is a doc comment for printing "3".
+                print("3")
+
+                print("4")
+                /** Inline comment. */ print("5")
+
+                print("6")// This is an important statement.
```

Comment:
So it looks like the two spaces after the semicolon in the original are the trailing trivia of the semicolon? I guess it's fine to discard that because the pretty printer will fix it up (and if there was a non-comment token here instead, the spaces wouldn't be trailing trivia, so it wouldn't be an issue). It might be worth a comment somewhere in the implementation (or the test?) calling this case out though as something we don't need to worry about.

Reply:
I added a comment in the implementation that the trailing trivia is being discarded. It seems that the trailing trivia is only spaces up to but not including the first newline or comment. The pretty printer will fix any missing spaces, so I didn't try to merge the trailing trivia into a different token.